monkey/monkey_island/cc/resources/pba_file_upload.py

```diff
@@ -96,16 +120,23 @@ def delete(self, target_os):
         if self._is_target_os_supported(target_os):
             return Response(status=HTTPStatus.UNPROCESSABLE_ENTITY, mimetype="text/plain")
 
-        filename_path = (
-            PBA_LINUX_FILENAME_PATH if target_os == "PBAlinux" else PBA_WINDOWS_FILENAME_PATH
-        )
-        filename = ConfigService.get_config_value(filename_path)
-        if filename:
+        original_agent_configuration = self._agent_configuration_repository.get_configuration()
+        self._update_config(target_os, "")
+
+        if target_os == LINUX_PBA_TYPE:
+            filename = original_agent_configuration.custom_pbas.linux_filename
+        else:
+            filename = original_agent_configuration.custom_pbas.windows_filename
+
+        try:
             self._file_storage_service.delete_file(filename)
-            ConfigService.set_config_value(filename_path, "")
+        except Exception as err:
+            # Roll back the entire transaction if part of it failed.
+            self._agent_configuration_repository.store_configuration(original_agent_configuration)
+            raise err
 
         # API Spec: HTTP status code should be 204
-        return make_response({}, 200)
+        return make_response({}, HTTPStatus.OK)
 
     @staticmethod
     def _is_target_os_supported(target_os: str) -> bool:
```

Comment:
We need to either change this to `_is_target_os_not_supported` or change the logic in this method to remove the `not` (and modify all related code in either case).

Reply:
Changed to `target_os_is_unsupported()`.